searx/plugins/oa_doi_rewrite.py

```diff
@@ -21,12 +26,20 @@ def extract_doi(url):
     return None
 
 
+def get_doi_resolver():
+    doi_resolvers = settings['doi_resolvers']
+    doi_resolver = request.args.get('doi_resolver', request.preferences.get_value('doi_resolver'))[0]
+    if doi_resolver not in doi_resolvers:
+        doi_resolvers = settings['default_doi_resolver']
+    return doi_resolvers[doi_resolver]
+
+
 def on_result(request, search, result):
     doi = extract_doi(result['parsed_url'])
     if doi and len(doi) < 50:
         for suffix in ('/', '.pdf', '/full', '/meta', '/abstract'):
             if doi.endswith(suffix):
                 doi = doi[:-len(suffix)]
-        result['url'] = 'http://doai.io/' + doi
+        result['url'] = get_doi_resolver() + doi
```

Comment:
You can pass `request.args` to `get_doi_resolver()` as a parameter from this function, this way you don't need the `from flask import request` line.